Algorithms-P3.playground/Contents.swift

```diff
@@ -55,7 +140,64 @@ class P3 {
      * Because "foo(bar(baz))blim" becomes "foo(barzab)blim" and then "foobazrabblim".
      */
     class func reverseInParentheses(_ inputString: String) -> String {
-        fatalError("not implemented")
+        var charInput = [Character](inputString)
```

Comment:
` class func reverseInParentheses(_ inputString: String) -> String {
        print(inputString)
        guard let closeIndex = inputString.firstIndex(of: ")") else {
            return inputString
        }
        
        let openIndex = inputString[..<closeIndex].lastIndex(of: "(")!
        
        let newString = inputString[inputString.startIndex..<openIndex] + inputString[inputString.index(after: openIndex)..<closeIndex].reversed() + inputString[inputString.index(after: closeIndex)..<inputString.endIndex]
        
        return reverseInParentheses(String(newString))
    }`